dispatcher.go

```diff
@@ -146,8 +167,9 @@ type dispatcher struct {
 
 	Name string
 
-	inbounds  Inbounds
-	outbounds Outbounds
+	inbounds   Inbounds
+	outbounds  Outbounds
+	transports map[transport.Transport]struct{}
```

Comment:
Seems like this should just be `[]transport.Transport` at this point - the map is useful for dedupping, but let's just call `keys` on it before passing it around.